lib/carma/client/mule_soft_client.rb

```diff
@@ -52,6 +58,39 @@ def raise_error_unless_success(resource, status)
 
         raise Common::Exceptions::SchemaValidationErrors, ["Expecting 200 status but received #{status}"]
       end
+
+      # New Authentication strategy
+      # Call Mulesoft with bearer token
+      def perform_post(payload)
```

Comment:
🎨 Is this going to replace `do_post` entirely after testing, or will both persist long-term? If the former, it won't much matter since we'll wind up with one `do_the_thing()` in this class, but if they're going to coexist for a while I'd prefer a more specific name. `do_post_with_bearer_token` or something, maybe?

Reply:
This will replace `do_post` entirely once we turn this on. That was why I named it more generically.